Receiving some errors on Windows.

Hey all,

Today I added myself as a contributor to one OSS repo through the all-contributors CLI. I ended up receiving some errors, even though it looked like the contributors file was updated and a commit was made.

Here is the full log:

```
joachim@DESKTOP-O4ENA28 E:\projects\codesandbox-client
> npm run add

> react-sandbox@0.1.0 add E:\projects\codesandbox-client
> all-contributors add

? What is the contributor's GitHub username? jseminck
? What are the contribution types? 💻  Code
undefined
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! react-sandbox@0.1.0 add: `all-contributors add`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the react-sandbox@0.1.0 add script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\joachim\AppData\Roaming\npm-cache\_logs\2017-08-18T15_35_23_293Z-debug.log

joachim@DESKTOP-O4ENA28 E:\projects\codesandbox-client
> cat C:\Users\joachim\AppData\Roaming\npm-cache\_logs\2017-08-18T15_35_23_293Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'add' ]
2 info using npm@5.3.0
3 info using node@v8.4.0
4 verbose run-script [ 'preadd', 'add', 'postadd' ]
5 info lifecycle react-sandbox@0.1.0~preadd: react-sandbox@0.1.0
6 info lifecycle react-sandbox@0.1.0~add: react-sandbox@0.1.0
7 verbose lifecycle react-sandbox@0.1.0~add: unsafe-perm in lifecycle true
8 verbose lifecycle react-sandbox@0.1.0~add: PATH: C:\Users\joachim\AppData\Roaming\nvm\v8.4.0\node_modules\npm\bin\node-gyp-bin;E:\projects\codesandbox-client\node_modules\.bin;E:\projects\apps\conemu\ConEmu\Scripts;E:\projects\apps\conemu;E:\projects\apps\conemu\ConEmu;C:\Program Files\Docker\Docker\Resources\bin;C:\ProgramData\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Skype\Phone\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Mercurial\;E:\projects\apps\apache-maven-3.3.9\bin;C:\Program Files\OpenVPN\bin;C:\Users\joachim\AppData\Roaming\nvm;C:\Program Files\nodejs;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Users\joachim\AppData\Local\Microsoft\WindowsApps;C:\Users\joachim\AppData\Roaming\npm;C:\Users\joachim\AppData\Local\atom\bin;C:\Users\joachim\AppData\Roaming\nvm;C:\Program Files\nodejs;C:\Users\joachim\AppData\Local\Microsoft\WindowsApps;;C:\Program Files (x86)\Microsoft VS Code\bin
9 verbose lifecycle react-sandbox@0.1.0~add: CWD: E:\projects\codesandbox-client
10 silly lifecycle react-sandbox@0.1.0~add: Args: [ '/d /s /c', 'all-contributors add' ]
11 silly lifecycle react-sandbox@0.1.0~add: Returned: code: 1  signal: null
12 info lifecycle react-sandbox@0.1.0~add: Failed to exec add script
13 verbose stack Error: react-sandbox@0.1.0 add: `all-contributors add`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Users\joachim\AppData\Roaming\nvm\v8.4.0\node_modules\npm\lib\utils\lifecycle.js:289:16)
13 verbose stack     at emitTwo (events.js:125:13)
13 verbose stack     at EventEmitter.emit (events.js:213:7)
13 verbose stack     at ChildProcess.<anonymous> (C:\Users\joachim\AppData\Roaming\nvm\v8.4.0\node_modules\npm\lib\utils\spawn.js:40:14)
13 verbose stack     at emitTwo (events.js:125:13)
13 verbose stack     at ChildProcess.emit (events.js:213:7)
13 verbose stack     at maybeClose (internal/child_process.js:927:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
14 verbose pkgid react-sandbox@0.1.0
15 verbose cwd E:\projects\codesandbox-client
16 verbose Windows_NT 10.0.15063
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "add"
18 verbose node v8.4.0
19 verbose npm  v5.3.0
20 error code ELIFECYCLE
21 error errno 1
22 error react-sandbox@0.1.0 add: `all-contributors add`
22 error Exit status 1
23 error Failed at the react-sandbox@0.1.0 add script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

If I installed the CLI manually, then I did not get any errors, just an `undefined` message:

```
joachim@DESKTOP-O4ENA28 E:\projects\codesandbox-client
> npm install all-contributors-cli -g
C:\Program Files\nodejs\all-contributors -> C:\Program Files\nodejs\node_modules\all-contributors-cli\cli.js
+ all-contributors-cli@4.4.0
added 144 packages in 21.045s

joachim@DESKTOP-O4ENA28 E:\projects\codesandbox-client
> all-contributors add
? What is the contributor's GitHub username? jseminck
? What are the contribution types? 💻  Code
undefined
```

Not sure, if it is related to me being on Windows... anyway, thought I'd report it in case anyone has had similar issues.

--- a/src/util/git.js
+++ b/src/util/git.js
@@ -37,8 +37,16 @@ function getRepoInfo() {
 
 const spawnGitCommand = pify((args, cb) => {
   const git = spawn('git', args)
-  git.stderr.on('data', cb)
-  git.on('close', cb)
+  const bufs = [];
+  git.stderr.on('data', (buf) => bufs.push(buf));
+  git.on('close', (code) => {
+    if (code) {
+      const msg = Buffer.concat(bufs).toString() || `git ${args.join('')} - exit code: ${code}`;
+      cb(new Error(msg));
+    } else {
+      cb(null);
+    }
+  });
 })
 
 function commit(options, data) {

